model/load_balancers_vms.rb

```diff
@@ -3,16 +3,75 @@
 require_relative "../model"
 
 class LoadBalancersVms < Sequel::Model
+  many_to_one :load_balancer
+  many_to_one :vm
   include ResourceMethods
+  include HealthMonitorMethods
+
+  def init_health_monitor_session
+    {
+      ssh_session: vm.vm_host.sshable.start_fresh_session
+    }
+  end
+
+  def health_check(session:)
+    [
+      check_probe(session, :ipv4),
+      check_probe(session, :ipv6)
+    ]
+  end
+
+  def check_probe(session, type)
+    if type == :ipv4
+      return "up" unless load_balancer.ipv4_enabled?
+    elsif type == :ipv6
+      return "up" unless load_balancer.ipv6_enabled?
+    else
+      raise "Invalid type: #{type}"
+    end
+
+    begin
+      (session[:ssh_session].exec!(health_check_cmd(type)).strip == "200") ? "up" : "down"
+    rescue
+      "down"
+    end
+  end
+
+  def health_check_cmd(type)
+    address = (type == :ipv4) ? vm.nics.first.private_ipv4.network : vm.ephemeral_net6.nth(2)
```

Comment:
we have a number of calls throughout our codebase that gets addresses of a vm by performing calls like
`vm.nics.first.private_ipv4.network`. in the future, should we encapsulate this in a separate method?

Reply:
Makes sense, I'll look to implement that in a separate patch.